src/OrchardCore.Modules/OrchardCore.Settings/Services/SetupEventHandler.cs

```diff
@@ -1,15 +1,15 @@
 using System;
-using System.Collections.Generic;
 using System.Threading.Tasks;
 using OrchardCore.Abstractions.Setup;
-using OrchardCore.Setup.Events;
+using OrchardCore.Setup;
+using OrchardCore.Setup.Services;
 
 namespace OrchardCore.Settings.Services
 {
     /// <summary>
     /// During setup, registers the Super User.
     /// </summary>
-    public class SetupEventHandler : ISetupEventHandler
+    public class SetupEventHandler : TenantSetupHandlerBase
```

Comment:
Should we then change the class name to `TenantSetupHandler`.

The problem is that here I prefer the old interface name ending with `EventHandler`.

So maybe `ITenantSetupEventHandler` and `TenantSetupEventHandlerBase` but looks too long.

Maybe worth that you ask to @sebastienros in the next triage if we could introduce a little breaking change as I suggested by keeping the `ISetupEventHandler` and just extending it with the new events with the help of a `SetupEventHandlerBase` class. And maybe also provide an extension marked as obsolete to be still able to call `Setup()` in place of `SetupAsync()`.

So that we have a clean interface by keeping the right name, no need to use a pragma and so on.

Reply:
I'll ask. But, with the current approach we do not need to introduce the breaking change.

Reply:
Yes, just to confirm because ATM I prefer the name of the old interface.

Reply:
@sebastienros what do you think here? Can we introduce a small breaking change? In summary we want to modify `ISetupEventHandler` interface from this

```C#
public interface ISetupEventHandler
{
    Task Setup(IDictionary<string, object> properties, Action<string, string> reportError);
}
```

to this

```C#
public interface ISetupEventHandler
{
    Task SetupAsync(SetupContext context);

    Task FailedAsync(SetupContext context);

    Task SucceededAsync();
}
```

This will break anyone that is using `ISetupEventHandler` in their project. This should be a low impact breaking change since `ISetupEventHandler` would not be widely used.

Reply:
Yes, I would go this way, much cleaner, but worth to wait for a confirmation.

If no feedback, let's still go this way ;)

Reply:
@jtkech Learned something new today! As it strange as it may sounds, but starting with C# 8.0, interfaces can have default implementation! Not a very good approach and we should avoiding doing this. But, it this case it make sense to use it to prevent breaking change. 

Check out the new changes. But I think it's pretty clean now.

Reply:
Okay cool then, I will look at it asap.